lib/puppet/util/windows/process.rb

```diff
@@ -8,8 +8,21 @@ module Puppet::Util::Windows::Process
 
   WAIT_TIMEOUT = 0x102
 
+  ERROR_DIRECTORY = 0x10B
   def execute(command, arguments, stdin, stdout, stderr)
-    Process.create( :command_line => command, :startup_info => {:stdin => stdin, :stdout => stdout, :stderr => stderr}, :close_handles => false )
+    args = {
+      :command_line => command,
+      :startup_info => {:stdin => stdin, :stdout => stdout, :stderr => stderr},
+      :close_handles => false
+    }
+    args[:cwd] = arguments[:cwd] if arguments[:cwd]
+
+    begin
+      Process.create(args)
+    rescue SystemCallError => e
+      raise e unless e.errno == ERROR_DIRECTORY && args[:cwd]
+      raise ArgumentError, _("Working directory %{cwd} does not exist!") % { cwd: args[:cwd] }
```

Comment:
Don't eat the existing exception here.

Also - does this handle access denied problems implicitly?

Reply:
You might also consider raising `Puppet::Util::Windows::Error` like this (depending on if your callers are OK with that error class)

```ruby
raise Puppet::Util::Windows::Error.new(_("Working directory %{cwd} does not exist!") % { cwd: args[:cwd] }, e.errno, e)
```

That helper exception type can lookup the Windows error message for the given win32 error code (which in this case may already be in the message for the `SystemCallError` - you'll have to check that)
https://github.com/puppetlabs/puppet/blob/ba4d1a1aba0095d3c70b98fea5c67434a4876a61/lib/puppet/util/windows/error.rb